kai/reactive_codeplanner/task_manager/api.py

```diff
@@ -29,6 +29,11 @@ class Task:
 
     _creation_counter = 0
 
+    def oldest_ancestor(self) -> "Task":
+        if self.parent:
+            return self.parent.oldest_ancestor()
+        return self
+
```

Comment:
Small performance thing. Are `Task`s immutable? If so, we could make this a cached property.

Reply:
hmm, they are not immutable, but the parent-child relationships are AFAIK. Though it'd probably save us a microsecond :joy:

Reply:
Hey those microseconds add up 👀